test/cql-pytest/test_describe.py

```diff
@@ -2567,3 +2568,35 @@ def aux(row: DescRow) -> DescRow:
             assert restore_stmts == res
         finally:
             cql.execute(f"DROP KEYSPACE IF EXISTS {ks}")
+
+### ===========================================================================
+
+# Verifies that restoring a UDF works correctly.
+# Reproduces: scylladb/scylladb#20256.
+def test_restore_udf(cql, test_keyspace):
+    with new_type(cql, test_keyspace, "(value int)") as udt:
+        _, udt_name = udt.split(".")
+
+        fn_content = f"""(lhs int, rhs {udt_name})
+            RETURNS NULL ON NULL INPUT
+            RETURNS {udt_name}
+            LANGUAGE lua
+            AS $$
+            rhs.value = lhs + rhs.value
+            return rhs
+            $$;"""
+        # Get rid of indentation in the string.
+        fn_content = inspect.cleandoc(fn_content)
```

Comment:
Unfortunately, we'd need to prepend the first line with whitespaces to be able to use `textwrap.dedent` or have an empty line at the beginning. The first solution would be ugly (in my opinion), the other would cause problems because DESCRIBE will not have the empty line and we'll end up having to modify the string again. A solution could be to create an instance of `textwrap.TextWrapper` with `drop_whitespace`, but it's simpler to use `inspect.cleandoc`.
If you think it's better to use one of the options or perhaps something else, I'm fine with it. For now I'll leave it as-is.

Reply:
> Unfortunately, we'd need to prepend the first line with whitespaces to be able to use `textwrap.dedent` or have an empty line at the beginning.

You can avoid this problem by using a backslash before the first newline:

```
>>> text = """\
...     Ala
...     ma
...     kota
...     """
>>> text
'    Ala\n    ma\n    kota\n    '
>>> import textwrap
>>> text = textwrap.dedent(text)
>>> text
'Ala\nma\nkota\n'
>>>
```

There is also a similar example in the [documentation entry](https://docs.python.org/3/library/textwrap.html#textwrap.dedent) for `textwrap.dedent`.